index.d.ts

```diff
@@ -1,12 +1,65 @@
-import { Parser } from 'acorn' 
+import * as acorn from 'acorn';
 
-declare const jsx: (options?: jsx.Options) => (BaseParser: typeof Parser) => typeof Parser;
+declare const jsx: {
+  tokTypes: typeof acorn.tokTypes;
+  (options?: jsx.Options): (BaseParser: typeof acorn.Parser) => jsx.AcornJsxParser
+}
 
 declare namespace jsx {
-  interface Options {
+
+  type tokTypes = typeof acorn.tokTypes;
+
+  export interface TokTypes extends tokTypes {
+    jsxName: acorn.TokenType,
+    jsxText: acorn.TokenType,
+    jsxTagEnd: acorn.TokenType,
+    jsxTagStart: acorn.TokenType
+  }
+
+  export interface Options {
```

Comment:
Types in a namespace are always exported, so the `export` keyword is redundant.

It’s not wrong, but personally I prefer to remove them.

Reply:
Ok. Applied. TS unfortunately isn't very helpful [in their docs](https://www.typescriptlang.org/docs/handbook/namespaces.html#ambient-namespaces) in helping give a better understanding of ambient namespaces like this; it seems to only speak there of browser global usage. I guess this may be because modules are moving to become the preferred way, and the fact that we still need to understand the old way when in use in CJS was lost along the way.